openedx/core/djangoapps/content/learning_sequences/api/outlines.py

```diff
@@ -84,6 +85,21 @@ def key_supports_outlines(opaque_key: OpaqueKey) -> bool:
     return False
 
 
+def public_api_available(course_key: CourseKey, user: types.User) -> bool:
+    """
+    Is the Public API availalbe for this Course to this User?
+
+    This only really exists while we do the waffle-flag rollout of this feature,
+    so that in-process callers from other apps can determine whether they should
+    trust Learning Sequences API data for a particular user/course.
+    """
+    return (
+        key_supports_outlines(course_key) and
+        LearningContext.objects.filter(context_key=course_key).exists() and
```

Comment:
Random thought: This check is an additional round trip to the database, which isn't a huge deal - but maybe the absence of a returned outline could be used instead for the LCs that don't yet exist?

Reply:
Yeah, I thought about that too. But in the end, I thought it'd be more convenient for external callers to deal with only two states: Yes, this user can use learning_sequences for this course, or No, this user can't use it. I think that most times they won't care about the specific reason why they can't use it.

Reply:
It's a simpler abstraction also - which almost always wins. 👍🏼